### PR TITLE
libusb1 - enable build with examples

### PR DESCRIPTION
**Pull request by i-am-logger**
I added `withExamples` to build all examples into `examples/bin`. 

since the `fxload` package is outdated and source code moved to  libusb's examples. I believe it is better to build `fxload` here and have it install to `/sbin` similarly to how `fxload` package does it. 

this version of `fxload` supports USB3 devices.

https://github.com/NixOS/nixpkgs/issues/207381

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -16085,4 +16085,10 @@
     github = "ziguana";
     githubId = 45833444;
   };
+  snick = {
```
Comment (prusnak) [inline maintainers/maintainer-list.nix:16088]: It's so much easier when this is the same as github nick ...

Also it seems the maintainers' list in this file is sorted, so please add at the correct position.

```suggestion
  realsnick = {
```

File: pkgs/development/libraries/libusb1/default.nix [hunk 2]
```diff
@@ -31,12 +33,23 @@ stdenv.mkDerivation rec {
...
+  configureFlags =
+    lib.optional (enableUdev) "--disable-udev"
+    ++ lib.optional (withExamples) "--enable-examples-build";
 
   preFixup = lib.optionalString enableUdev ''
     sed 's,-ludev,-L${lib.getLib udev}/lib -ludev,' -i $out/lib/libusb-1.0.la
   '';
 
+  preInstall = lib.optionalString withExamples ''
+    mkdir -p $out/{bin,sbin,examples/bin}
```
Comment (prusnak) [inline pkgs/development/libraries/libusb1/default.nix:45]: Does this really need to be in `preInstall`? Maybe it's enough to have this line as the first line of `postInstall`?

File: pkgs/development/libraries/libusb1/default.nix [hunk 3]
```diff
@@ -45,6 +58,10 @@ stdenv.mkDerivation rec {
     '';
     platforms = platforms.all;
     license = licenses.lgpl21Plus;
-    maintainers = with maintainers; [ prusnak ];
+    maintainers = with maintainers; [ prusnak snick ];
   };
 }
+
```
Comment (prusnak) [inline pkgs/development/libraries/libusb1/default.nix:64]: Remove these empty lines

File: pkgs/development/libraries/libusb1/default.nix [hunk 4]
```diff
@@ -31,12 +33,23 @@ stdenv.mkDerivation rec {
 
   dontDisableStatic = withStatic;
 
-  configureFlags = lib.optional (!enableUdev) "--disable-udev";
+  configureFlags =
+    lib.optional (enableUdev) "--disable-udev"
```
Comment (prusnak) [inline pkgs/development/libraries/libusb1/default.nix:37]: You missed the `!` when rewriting this line

```suggestion
    lib.optional (!enableUdev) "--disable-udev"
```

**Review by prusnak — CHANGES_REQUESTED** [review]
Please address the review suggestions

File: pkgs/development/libraries/libusb1/default.nix (hunk 2 above)
Comment (realsnick) [reply pkgs/development/libraries/libusb1/default.nix:45]: i'll move to postInstall

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (realsnick) [reply maintainers/maintainer-list.nix]: the maintainers list isn't sorted...just happen to look like that on few recent names

**Comment by realsnick** [comment]
thank you @prusnak , made the changes. Just didn't change the maintainr list, it is not sorted. Happy Holidays!

**Comment by prusnak** [comment]
Can you please split the change in `maintainers/maintainer-list.nix` into the commit with commit message: ``

Can you please update the commit message to: `maintainers: add realsnick`.

The second commit message should read: `libusb1: add option to build with examples`

**Comment by realsnick** [comment]
> Can you please split the change in `maintainers/maintainer-list.nix` into a separate commit with commit message: `maintainers: add realsnick`?
> 
> The remaining changes should be put into a commit with message: `libusb1: add option to build with examples`

done. thank you!

**Comment by prusnak** [comment]
@NixOS/ofborg eval

**Comment by prusnak** [comment]
@GrahamcOfBorg eval

**Comment by prusnak** [comment]
Hm, why does this change causes rebuild?

**Comment by raphaelr** [comment]
> Hm, why does this change causes rebuild?

This adds a new derivation attribute, `postInstall = ""`.

**Comment by prusnak** [comment]
> This adds a new derivation attribute, `postInstall = ""`.

Is there a way how to achieve the same but not trigger rebuilds?

**Comment by realsnick** [comment]
@prusnak by triggering rebuilds you mean it will rebuild everything that depends on libusb1? 

If that the case, is there a way to put the lib.option withExamples outside the event handlers? sorry i'm new to nix and its syntax

**Comment by raphaelr** [comment]
As far as I know, there's no idiomatic way to prevent the mass rebuild here.

That's ok though, just retarget this PR to the `staging` branch. Please follow the instructions in  [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging) on how to do this correctly without causing a mass ping.

**Comment by realsnick** [comment]
Should I recreate the PR to the staging branch?

**Pull request closed without merging**